front/context/src/ApiContext.tsx

```diff
@@ -19,43 +19,63 @@ export const ApiContext: React.Context<ApiContextType> = React.createContext(
   {} as ApiContextType
 );
 
+export interface ApiType {
+  withPromise?: boolean;
+  withRxjs?: boolean;
+}
+
 export interface ApiContextProviderProps {
   children?: React.ReactElement;
   provider: ProviderInterface;
-  promise?: boolean;
+  type?: ApiType;
 }
 
 export function ApiContextProvider(
   props: ApiContextProviderProps
 ): React.ReactElement {
-  const { children = null, provider, promise } = props;
+  const {
+    children = null,
+    provider,
+    type = { withRxjs: true, withPromise: false },
+  } = props;
   const [api, setApi] = useState<ApiRx>(new ApiRx({ provider }));
   const [apiPromise, setApiPromise] = useState<ApiPromise>(
     new ApiPromise({ provider })
```

Comment:
This is where the decoration happens (can be a bit heavy on UI), so we need to `new` them only if the corresponding `with` is set